src/driver/open.rs

```diff
@@ -6,24 +6,25 @@ use std::io;
 use std::path::Path;
 
 /// Open a file
+#[allow(dead_code)]
 pub(crate) struct Open {
-    pub(crate) _path: CString,
+    pub(crate) path: CString,
+    pub(crate) flags: libc::c_int,
 }
 
 impl Op<Open> {
     /// Submit a request to open a file.
-    pub(crate) fn open(path: &Path, options: &OpenOptions) -> io::Result<Op<Open>> {
+    pub(crate) fn open(path: impl AsRef<Path>, options: &OpenOptions) -> io::Result<Op<Open>> {
```

Comment:
Normally I'd say this should be its own PR, but I don't have a problem leaving it in this one right now.